### PR TITLE
Sync all compiler exercise instructions

### PR DESCRIPTION
**Pull request by bernhardmgruber**
~I am really confused now. The slides say the compiler exercise uses the `functions` exercise. But the `polymorphism` exercise has the instructions for the exercise. Which one is not the right one?~

The compiler exercise is now done again on the polymorphism exercise. This was changed in 696cd88 from PR #88 to the functions exercise, so the tools can be discussed even if the OO part was not discussed yet. However, the polymorphism exercise is better since it involves a shared library.

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
I synchronized all instructions again and use the polymorphism exercise. PR description updated.

**Review by sponce — APPROVED** [review]

File: talk/tools/compiling.tex [hunk 1]
```diff
@@ -212,19 +212,19 @@
...
     \item see linking stage of the final program using g++ -v
       \begin{itemize}
-      \item just add a -v in the Makefile command for functions target
+      \item just add a -v in the Makefile command for trypoly target
       \item run make clean; make
-      \item look at the collect 2 line, from the end up to ``-o functions''
+      \item look at the collect 2 line, from the end up to ``-o trypoly''
       \end{itemize}
-    \item see library dependencies with `ldd functions`
+    \item see library dependencies of `trypoly`
```
Comment (hageboeck) [inline talk/tools/compiling.tex:227]:
```suggestion
    \item see library dependencies of `trypoly` using `ldd`
```

File: code/polymorphism/README.md [hunk 2]
```diff
@@ -20,13 +20,13 @@ Step 2
...
 * try `make clean; make`
 * see linking stage using `g++ -v`
-  * just add a -v in the Makefile command for trypoly
-  * run make clean; make
-  * look at the collect 2 line, from the end up to '-o trypoly'
-* see library dependencies with `ldd`
+  * just add a `-v` in the Makefile command for `trypoly`
+  * run `make clean; make`
+  * look at the collect 2 line, from the end up to `-o trypoly`
+* see library dependencies of `trypoly` and `Polygons.so` with `ldd`
```
Comment (hageboeck) [inline code/polymorphism/README.md:32]:
```suggestion
* see library dependencies of `trypoly` and `Polygons.so` with `ldd`
  * On OS X, use `otool -L` for a similar effect
```

**Review by hageboeck — APPROVED** [review]
LGTM, but maybe mention ldd and otool -L

**Pull request merged**